drivers/modem/quectel-bg9x.c

```diff
@@ -953,6 +953,57 @@ static const struct setup_cmd setup_cmds[] = {
 	SETUP_CMD_NOHANDLE("AT+QICSGP=1,1,\"" MDM_APN "\",\"" MDM_USERNAME "\", \"" MDM_PASSWORD "\",1"),
 };
 
+
+/* Func: modem_pdp_context_open
+ * Desc: This helper function is called from modem_setup, and is
+ * used to open the PDP context. If there is trouble opening the
+ *  PDP context, we try to deactive and reactive 3 times.
+ *  If it fails, we return an error.
+ */
+
+static int modem_pdp_context_open(int contextID)
+{
+
+	int ret;
+	int pdp_act_retry_count = 0;
+	char buf_act[sizeof("AT+QIACT=##")] = {0};
+	char buf_deact[sizeof("AT+QIDEACT=##")] = {0};
+
+	snprintk(buf_act, sizeof(buf_act), "AT+QIACT=%d", contextID);
+	snprintk(buf_deact, sizeof(buf_deact), "AT+QIDEACT=%d", contextID);
+
+	ret = modem_cmd_send(&mctx.iface, &mctx.cmd_handler,
+			     NULL, 0U, buf_act, &mdata.sem_response,
+			     MDM_CMD_TIMEOUT);
+
+	/* If there is trouble opening the PDP context, we try to deactivate/reactive it. */
+	while (ret == -EIO && pdp_act_retry_count < MDM_PDP_ACT_RETRY_COUNT) {
+
+		ret = modem_cmd_send(&mctx.iface, &mctx.cmd_handler,
+			     NULL, 0U, buf_deact, &mdata.sem_response,
+			     MDM_CMD_TIMEOUT);
+
+		/* If there's an error for the AT+QIDEACT, restart the module. */
+		if (ret < 0) {
+			return -1;
```

Comment:
why return -1 and not ret ?

Reply:
Fixed. Returned -ret because the error number is a positive one.